website/client/components/tasks/column.vue

```diff
@@ -591,35 +567,36 @@ export default {
         }
       });
     },
-    filterTask (task) {
-      // View
-      if (!this.activeFilters[task.type].filter(task)) return false;
-
-      // Tags
-      const selectedTags = this.selectedTags;
-
-      if (selectedTags && selectedTags.length > 0) {
-        const hasAllSelectedTag = selectedTags.every(tagId => {
-          return task.tags.indexOf(tagId) !== -1;
-        });
-
-        if (!hasAllSelectedTag) return false;
+    filterByTagList (taskList, tagList = []) {
+      let filteredTaskList = taskList;
+      // fitler requested tasks by tags
+      if (!isEmpty(tagList)) {
+        filteredTaskList = taskList.filter(
+          task => tagList.every(tag => task.tags.indexOf(tag) !== -1)
+        );
       }
-
-      // Text
-      const searchText = this.searchText;
-
-      if (!searchText) return true;
-      if (task.text.toLowerCase().indexOf(searchText) !== -1) return true;
-      if (task.notes.toLowerCase().indexOf(searchText) !== -1) return true;
-
-      if (task.checklist && task.checklist.length) {
-        const checklistItemIndex = task.checklist.findIndex(({text}) => {
-          return text.toLowerCase().indexOf(searchText) !== -1;
-        });
-
-        return checklistItemIndex !== -1;
+      return filteredTaskList;
+    },
+    filterBySearchText (taskList, searchText = '') {
+      let filteredTaskList = taskList;
+      // filter requested tasks by search text
+      if (!isEmpty(searchText)) {
+        // to ensure broadest case insensitive search matching
+        let searchTextLowerCase = searchText.toLowerCase();
+        filteredTaskList = taskList.filter(
+          task => {
+            // eslint rule disabled for block to allow nested binary expression
+            /* eslint-disable no-extra-parens */
+            return (
+              (!isEmpty(task.text) && task.text.toLowerCase().indexOf(searchTextLowerCase) > -1) ||
```

Comment:
why ar you using `!isEmpty` instead of `task.note && ...` these are simple strings, not objects

Reply:
And the check for `task.text` isn't necessary since we enforce its existence

Reply:
Force of habit trying to be thorough.

Reply:
I tend to go overboard sometimes, thanks for the heads up